packages/sandbox/src/file_watching_sandbox.ts

```diff
@@ -152,7 +152,7 @@ export class FileWatchingSandbox extends EventEmitter implements Sandbox {
       this.emitWatching();
     });
 
-    if (options.watchForChanges) {
+    if (options.watchForChanges == null || options.watchForChanges) {
```

Comment:
this is counter intuitive, if `watchForChanges` is null, why do we start the watcher?

Reply:
I flipped the flag to be a "positive" boolean in response to "https://github.com/aws-amplify/amplify-backend/pull/1298#discussion_r1566238328.

In case where the option is not defined, I wanted to keep the default behavior to start the watcher. We should only prevent the watcher from starting if the `watchForChanges` is explicitly set to false.

I can update this to `if (options.watchForChanges !== false)` for better readbility,

Reply:
Or you can add the default for `watchForChanges` as true for Sandbox.